### PR TITLE
Dual video

### PR DESCRIPTION
**Pull request by skywilliamsonl**
Summary: Enable multiple media pipeline creation from the rialto sinks, and group sinks with the same parent gst bin to a single media pipeline.
Type: Feature
Test Plan: Youtube Certification Tests
Dependencies and Impacts: None
Jira: RIALTO-10

### CODE REVIEW

**Review by aczs — APPROVED** [review]

File: source/RialtoGStreamerMSEVideoSink.cpp [hunk 1]
```diff
@@ -46,15 +48,28 @@ enum
...
+        // maxWidth and maxHeight are used to set the video capabilities of the MediaPlayer.
+        // If the mediaPlayer has already been created (ie. an audio sink on the same parent bus changed state first)
+        // the video capabilities will NOT be set.
+        if (!basePriv->m_mediaPlayerManager.attachMediaPlayerClient(rialto_mse_base_get_oldest_gst_bin_parent(element),
+                                                                    priv->maxWidth, priv->maxHeight))
+        {
+            GST_ERROR_OBJECT(sink, "Cannot attach the MediaPlayerClient");
+            return GST_STATE_CHANGE_FAILURE;
+        }
+
```
Comment (aczs) [inline source/RialtoGStreamerMSEVideoSink.cpp:69]: Can you add info log here? Something like:
```
GstObject parentObject = rialto_mse_base_get_oldest_gst_bin_parent(element);
if (!basePriv->m_mediaPlayerManager.attachMediaPlayerClient(parentObject, priv->maxWidth, priv->maxHeight))
    {
        GST_ERROR_OBJECT(sink, "Cannot attach the MediaPlayerClient");
        return GST_STATE_CHANGE_FAILURE;
    }
    else
   {
       GST_INFO_OBJECT(element, "Attached media player client with parent %s(%p)", gst_object_get_name(parentObject), parentObject )
   }
```

**Review by aczs — DISMISSED** [review]

**Review by skywojciechowskim — APPROVED** [review]

File: source/RialtoGStreamerMSEVideoSink.cpp (hunk 1 above)
Comment (skywilliamsonl) [reply source/RialtoGStreamerMSEVideoSink.cpp:74]: Done

File: source/RialtoGStreamerMSEAudioSink.cpp [hunk 2]
```diff
@@ -45,8 +45,21 @@ static GstStateChangeReturn rialto_mse_audio_sink_change_state(GstElement *eleme
     {
     case GST_STATE_CHANGE_READY_TO_PAUSED:
     {
+        // Attach the media player client to media player manager
+        GstObject parentObject = rialto_mse_base_get_oldest_gst_bin_parent(element);
```
Comment (aczs) [inline source/RialtoGStreamerMSEAudioSink.cpp:49]: I haven't checked if code which I proposed compiles. I think that parentObject should be a pointer.

File: source/MediaPlayerManager.cpp [hunk 3]
```diff
@@ -17,95 +17,158 @@
...
+        releaseMediaPlayerClient();
+        createMediaPlayerClient(gstBinParent, maxVideoWidth, maxVideoHeight);
+    }
+
+    if (!m_client.lock())
+    {
+        GST_ERROR("Failed to attach the media player client");
+        return false;
+    }
+    else
```
[the remark this answers is not in the document]
Comment (skywilliamsonl) [reply source/MediaPlayerManager.cpp:51]: Done

File: source/RialtoGStreamerMSEAudioSink.cpp (hunk 2 above)
Comment (skywilliamsonl) [reply source/RialtoGStreamerMSEAudioSink.cpp]: Yes it compiles now :D

**Review by aczs — DISMISSED** [review]

**Review by skywojciechowskim — DISMISSED** [review]

**Review by aczs — APPROVED** [review]

**Pull request merged**